src/txkube/testing/strategies.py

```diff
@@ -71,7 +71,17 @@ def dns_labels():
     ))
 
 # XXX wrong
-object_name = object_names = image_names = dns_labels
+object_name = object_names = dns_labels
+
+
+def image_names():
+    """
+    Strategy to build Docker image names.
+
+    Only generate images that appear to be hosted on localhost to avoid ever
+    actually pulling an image from anywhere on the network.
+    """
+    return joins(u"/", tuples(just(u"127.0.0.1"), dns_labels()))
```

Comment:
It seems like joins existed already, but I find this a bit less clear than just

return dns_labels().map(lambda label: "127.0.0.1/" + label)